### PR TITLE
Fix Python encoding errors

### PR DESCRIPTION
**Pull request by jwitchard-r**
The original xplenty package wasn't geared for Python3 which requires byte input for some urllib functionality instead of strings. This PR adds conversions from strings to byte objects, and adds the appropriate authorization header encodings ([see stackoverflow](https://stackoverflow.com/questions/24635064/how-to-use-urllib-in-python-3/24648149#24648149))

As for parsing the response, no changes were required as json.loads accepts str|byte input

### CODE REVIEW

File: xplenty/xplenty_api.py [hunk 1]
```diff
@@ -305,94 +304,100 @@ def delete(self, url):
...
     def _join_url(self, method):
-        _url = API_URL % ( self.account_id )
-        url = urljoin(_url , method )
+        _url = API_URL % (self.account_id)
+        url = urljoin(_url, method)
         return url
 
+    def _add_authorization_header(self, request, api_key: str):
+        # api_key_bytes = ('%s' % api_key).encode('utf-8')
+        # base64string = base64.encodebytes(api_key_bytes).rstrip()
```
Comment (jrmay) [inline xplenty/xplenty_api.py]: Any reason to keep this commented code?

**Review by jrmay — APPROVED** [review]
Is is correct that the only net change here is to _stop_ base64 encoding payloads?

File: xplenty/xplenty_api.py [hunk 2]
```diff
@@ -305,94 +304,99 @@ def delete(self, url):
...
-        _url = API_URL % ( self.account_id )
-        url = urljoin(_url , method )
+        _url = API_URL % (self.account_id)
+        url = urljoin(_url, method)
         return url
 
+    def _add_authorization_header(self, request, api_key: str):
+        base64string = base64.b64encode(bytes('%s' % api_key, 'ascii'))
+        request.add_header("Authorization", "Basic %s" % base64string.decode('utf-8'))
+
```
Comment (toddnestor) [inline xplenty/xplenty_api.py:314]: Reading the documentation it seems they would expect base 64 encoded, and have it be the api key followed by a colon.

https://github.com/xplenty/xplenty-api-doc-v2#authentication

But does this work as is? If you functionally tested and this works then cool

**Review by toddnestor — APPROVED** [review]
LGTM pending response to one question.

File: xplenty/xplenty_api.py (hunk 2 above)
Comment (jwitchard-r) [reply xplenty/xplenty_api.py:314]: I did finally test this locally 🤭 , it was working without the colon and thats how the original xplenty-python-api was written but I added it anyhow (works too).

**Pull request merged**